Mathlib/Probability/Density.lean

```diff
@@ -136,6 +137,17 @@ theorem lintegral_eq_measure_univ {X : Ω → E} [HasPDF X ℙ μ] :
     Measure.map_apply (HasPDF.measurable X ℙ μ) MeasurableSet.univ, Set.preimage_univ]
 #align measure_theory.pdf.lintegral_eq_measure_univ MeasureTheory.pdf.lintegral_eq_measure_univ
 
+theorem unique [IsFiniteMeasure ℙ] {X : Ω → E} [HasPDF X ℙ μ] (f : E → ℝ≥0∞)
+    (hmf : AEMeasurable f μ) : ℙ.map X = μ.withDensity f ↔ pdf X ℙ μ =ᵐ[μ] f := by
+  rw [map_eq_withDensity_pdf X ℙ μ]
+  apply AEMeasurable.withDensity_eq_iff (measurable_pdf X ℙ μ).aemeasurable hmf
+  rw [lintegral_eq_measure_univ]
+  exact (measure_lt_top _ _).ne
+
+theorem unique' [SigmaFinite μ] {X : Ω → E} [HasPDF X ℙ μ] (f : E → ℝ≥0∞) (hmf : Measurable f) :
```

Comment:
This will be generalized to `AEMeasurable` as well, and then you can delete the finite measure one.

Reply:
Indeed you're right. I did not read carefully enough and thought the two conditions were on the same measure.